### PR TITLE
Hallway textscreens

### PR DESCRIPTION
**Pull request by graevy**
## About the PR

- extended the shuttle timer screens to accept comms console text broadcasts
- changed the comms console UI to accommodate this (first UI change; please critique hard)
- made the screens interactable, wall-snapping, clickable (sorry. forgot this in the original)
- fixed a bug in `roundendsystem.GetCentcomm` that i missed earlier (always returned null iirc)
- renamed the server `shuttletimersystem` etc into just `screensystem` etc

## Technical details

- the comms console broadcast button uses the announcement button's cooldown. i added a broadcast button cooldown but it didn't seem useful
- added `DefaultText` to `TextScreenVisuals` which stores the fallback text that gets rendered after timers are completed. this is only really relevant if the shuttle gets recalled
- client textscreensystem appearance change event handler now only checks the `AppearanceChangeEvent` args dict for updates
- reworked the server screensystem to only pass around regular strings. `string?[]` conversion is clientside
- added `devicenetworkcomponent` to the comms console proto
- added a 1/1000 chance for the evac shuttle docking to change the screen texts to a red KILL. requested by 2 art directors don't @ me

## Media

dotnet_ylEYJcSEWH

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Added hallway screen prototypes and comms console text broadcasts.

### CODE REVIEW

**Comment by UbaserB** [comment]
yooooooo

**Comment by Tayrtahn** [comment]
Ooo, cool!

Suggestion: match the text color to the station alert level

**Comment by Yuoko** [comment]
Hell yeah! This is good PR!

File: Content.Server/Shuttles/Systems/EmergencyShuttleSystem.cs [hunk 1]
```diff
@@ -235,8 +236,18 @@ private void OnEmergencyFTLComplete(EntityUid uid, EmergencyShuttleComponent com
                 [ShuttleTimerMasks.ShuttleTime] = countdownTime,
                 [ShuttleTimerMasks.SourceTime] = countdownTime,
                 [ShuttleTimerMasks.DestTime] = countdownTime,
-                [ShuttleTimerMasks.Text] = new string?[] { "BYE!" }
             };
+
+            // by popular request
+            // https://discord.com/channels/310555209753690112/770682801607278632/1189989482234126356
+            if (new Random().Next(1000) == 0)
```
Comment (juliangiebel) [inline Content.Server/Shuttles/Systems/EmergencyShuttleSystem.cs:243]: Please use `IRobustRandom` to generate a random number. It's already a dependency of this system and called `_random`

File: Resources/Prototypes/Entities/Structures/Machines/Computers/computers.yml [hunk 2]
```diff
@@ -548,6 +548,9 @@
     access: [[ "Command" ]]
   - type: CommunicationsConsole
     title: comms-console-announcement-title-station
+  - type: DeviceNetwork
+    transmitFrequency: 2450
```
Comment (juliangiebel) [inline Resources/Prototypes/Entities/Structures/Machines/Computers/computers.yml:552]: You don't need to set the requency if you're using `transmitFrequencyId`

**Review by juliangiebel — CHANGES_REQUESTED** [review]
Really neat PR 👍.
I found some stuff that needs to be changed.

**Comment by graevy** [comment]
> Ooo, cool!
> 
> Suggestion: match the text color to the station alert level

i really like this, though i think green is worse than /tg/ blue. @EmoGarbage404 @Flareguy it's pretty trivial to implement how do we feel

File: Content.Server/Screens/Systems/ScreenSystem.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            && text != null
+            )
+            _appearanceSystem.SetData(uid, TextScreenVisuals.DefaultText, text);
+    }
+
+    /// <summary>
+    /// Determines if/how a timer broadcast packet affects this screen.
+    /// Currently there are 2 broadcast domains: Arrivals, and every other screen.
+    /// All timer packets are broadcast in their domain. Subdomains are the shuttle, source, and dest.
+    /// Subdomains are linked by MapUid. Source/dest change each jump.
```
[the remark this answers is not in the document]
Comment (graevy) [reply Content.Server/Screens/Systems/ScreenSystem.cs]: i've changed it to:

```
/// <summary>
/// Determines if/how a timer packet affects this screen.
/// Currently there are 2 broadcast domains: Arrivals, and every other screen.
/// Domain is determined by the <see cref="DeviceNetworkComponent.TransmitFrequencyId"/> on each timer.
/// Each broadcast domain is divided into subnets. Screen MapUid determines subnet.
/// Subnets are the shuttle, source, and dest. Source/dest change each jump.
/// This is required to send different timers to the shuttle/terminal/station.
/// </summary>
```

does this do a better job?

**Comment by graevy** [comment]
> Really neat PR 👍. I found some stuff that needs to be changed.

thank you for the helpful feedback! i agree with all of it.

**Comment by Flareguy** [comment]
> > Ooo, cool!
> > Suggestion: match the text color to the station alert level
> 
> i really like this, though i think green is worse than /tg/ blue. @EmoGarbage404 @Flareguy it's pretty trivial to implement how do we feel

Feel free to try it if you want, it's probably not a good idea though as the blue really helps the entire sprite contrast well. I can't imagine red alert looking that good, especially.

**Pull request merged**